Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -816,36 +774,74 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
-                    let query = "SELECT collection_id_blob_base64, package_repository_url, name FROM \(Self.targetsFTSName);"
-                    try self.executeStatement(query) { statement in
-                        while let row = try statement.step() {
-                            if self.isShuttingdown.get() ?? false { return }
+    private func createSchemaIfNecessary(db: SQLite) throws {
+        let table = """
+            CREATE TABLE IF NOT EXISTS \(Self.packageCollectionsTableName) (
+                key STRING PRIMARY KEY NOT NULL,
+                value BLOB NOT NULL
+            );
+        """
+        try db.exec(query: table)
 
-                            let targetName = row.string(at: 2)
+        do {
+            let ftsPackages = """
+                CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.packagesFTSName) USING fts4(
+                    collection_id_blob_base64, id, version, name, repository_url, summary, keywords, products, targets,
+                    notindexed=collection_id_blob_base64,
+                    tokenize=unicode61
+                );
+            """
+            try db.exec(query: ftsPackages)
 
-                            if let collectionData = Data(base64Encoded: row.string(at: 0)),
-                                let collection = try? self.decoder.decode(Model.CollectionIdentifier.self, from: collectionData) {
-                                let collectionPackage = CollectionPackage(collection: collection, package: PackageIdentity(url: row.string(at: 1)))
-                                self.targetTrie.insert(word: targetName.lowercased(), foundIn: collectionPackage)
-                            }
-                        }
-                    }
-                    callback(.success(()))
-                    return true
-                } catch {
-                    callback(.failure(error))
-                    return false
-                }
-            }
+            let ftsTargets = """
+                CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.targetsFTSName) USING fts4(
+                    collection_id_blob_base64, package_repository_url, name,
+                    notindexed=collection_id_blob_base64,
+                    tokenize=unicode61
+                );
+            """
+            try db.exec(query: ftsTargets)
+
+            useSearchIndices.put(true)
+        } catch {
+            // We can use FTS3 tables but queries yield different results when run on different
+            // platforms. This could be because of SQLite version perhaps? But since we can't get
+            // consistent results we will not fallback to FTS3 and just give up if FTS4 is not available.
+            useSearchIndices.put(false)
         }
+
+        try db.exec(query: "PRAGMA journal_mode=WAL;")
+    }
+
+    private func withStateLock<T>(_ body: () throws -> T) throws -> T {
+        try self.stateLock.withLock(body)
+        /*switch self.location {
```

Comment:
delete?

Reply:
still thinking about this one... in manifest loader we use a FS lock since we ran into contention issues on Linux (my guess is that the sqlite version we use on linux is not compiled with the right flags)